src/replica/replica_disk_migrator.cpp

```diff
@@ -148,12 +152,140 @@ bool replica_disk_migrator::check_migration_args(const replica_disk_migrate_requ
     return true;
 }
 
-// TODO(jiashuo1)
 // THREAD_POOL_REPLICATION_LONG
-void replica_disk_migrator::migrate_replica(const replica_disk_migrate_request &req) {}
+void replica_disk_migrator::do_disk_migrate_replica(const replica_disk_migrate_request &req)
+{
+    if (status() != disk_migration_status::MOVING) {
+        derror_replica("disk migration(origin={}, target={}), err = Invalid migration status({})",
+                       req.origin_disk,
+                       req.target_disk,
+                       enum_to_string(status()));
+        reset_status();
+        return;
+    }
+
+    if (init_target_dir(req) && migrate_replica_checkpoint(req) && migrate_replica_app_info(req)) {
+        _status = disk_migration_status::MOVED;
+        ddebug_replica("disk migration(origin={}, target={}), update status from {}=>{}, ready to "
+                       "close origin replica({})",
+                       req.origin_disk,
+                       req.target_disk,
+                       enum_to_string(disk_migration_status::MOVING),
+                       enum_to_string(status()),
+                       _replica->dir());
+
+        close_current_replica();
+    }
+}
+
+// THREAD_POOL_REPLICATION_LONG
+bool replica_disk_migrator::init_target_dir(const replica_disk_migrate_request &req)
+{
+    // replica_dir: /root/origin_disk_tag/gpid.app_type
+    std::string replica_dir = _replica->dir();
+    // using origin dir to init new dir
+    boost::replace_first(replica_dir, req.origin_disk, req.target_disk);
+    if (utils::filesystem::directory_exists(replica_dir)) {
+        derror_replica("migration target replica dir({}) has existed", replica_dir);
+        reset_status();
+        return false;
```

Comment:
So how does the user know the result that failure happened?

Reply:
copy data is designed async and no response the status, the `async process` now need check the log

Reply:
or I can add `query_status` rpc in later pr?

Reply:
I agree with @neverchanje, there should be a way to let user notice failure happened, `query_status` rpc may be helpful.

Reply:
I can add later pr

Reply:
So what is your plan? You certainly can not actively notify the user. One possible way is to save the error, and every time user queries the status, if failure happened, you respond with that error.

Reply:
Right, now has one way is run `disk-migrate -g gpid`, if has one task is running, it will show:
```
 Internal server error [DiskMigrate to session 127.0.0.1:45801(replica)] failed: ErrorCode({Errno:ERR_BUSY}):Existed migrate task(replication::disk_migration_status::MOVED) is running]
```
We can get the task process but not need `query_status`